examples/events_example.py

```diff
@@ -0,0 +1,175 @@
+
+
+@coc.ClanEvents.member_received()
+async def on_clan_member_donation_receive(old_member, new_member):
+    final_received_troops = new_member.received - old_member.received
+    log.info(
+        f"{new_member} of {new_member.clan} just received {final_received_troops} troops.")
+
+
+@coc.ClanEvents.member_join()
+async def on_clan_member_join(member, clan):
+    log.info(f"{member.name} has joined {clan.name}")
+
+
+@coc.ClanEvents.member_leave()
+async def on_clan_member_leave(member, clan):
+    log.info(f"{member.name} has left {clan.name}")
+
+
+@coc.ClanEvents.points()
+async def on_clan_trophy_change(old_clan, new_clan):
+    log.info(
+        f"{new_clan.name} total trophies changed from {old_clan.points} to {new_clan.points}")
+
+
+@coc.ClanEvents.member_versus_trophies()
+async def clan_member_versus_trophies_changed(old_member, new_member):
+    log.info(
+        f"{new_member} versus trophies changed from {old_member.versus_trophies} to {new_member.versus_trophies}")
+
+
+"""War Events"""
+
+
+@coc.WarEvents.war_attack()
+async def current_war_stats(attack, war):
+    log.info(
+        f"Attack number {attack.order}\n({attack.attacker.map_position}).{attack.attacker} of {attack.attacker.clan} "
+        f"attacked ({attack.defender.map_position}).{attack.defender} of {attack.defender.clan}")
+
+
+"""Player Events"""
+
+
+@coc.PlayerEvents.donations()
+async def on_player_donation(old_player, new_player):
+    final_donated_troops = new_player.donations - old_player.donations
+    log.info(
+        f"{new_player} of {new_player.clan} just donated {final_donated_troops} troops.")
+
+
+@coc.PlayerEvents.received()
+async def on_player_donation_receive(old_player, new_player):
+    final_received_troops = new_player.received - old_player.received
+    log.info(
+        f"{new_player} of {new_player.clan} just received {final_received_troops} troops.")
+
+
+@coc.PlayerEvents.trophies()
+async def on_player_trophy_change(old_player, new_player):
+    log.info(
+        f"{new_player} trophies changed from {old_player.trophies} to {new_player.trophies}")
+
+
+@coc.PlayerEvents.versus_trophies()
+async def on_player_versus_trophy_change(old_player, new_player):
+    log.info(
+        f"{new_player} versus trophies changed from {old_player.trophies} to {new_player.trophies}")
+
+
+"""Client Events"""
+
+
+@coc.ClientEvents.maintenance_start()
+async def on_maintenance():
+    log.info("Maintenace Started")
+
+
+@coc.ClientEvents.maintenance_completion()
+async def on_maintenance_completion(time_started):
+    log.info("Maintenace Ended; started at %s", time_started)
+
+
+@coc.ClientEvents.new_season_start()
+async def season_started():
+    log.info("New season started, and will finish at %s",
+             str(utils.get_season_end()))
+
+
+
+async def main() -> None:
+    # Instantiate the coc_client using the Events client
+    coc_client = coc.EventsClient(
```

Comment:
Again, if this was defined at the top of the file we could use @coc_client.event all the way through 🙂

As I'm going through, I'm wondering what the easiest (and intuitive) way of registering events is. I'm not sure. Maybe by the end I'll have some more ideas